src/main/java/io/cdap/delta/bigquery/MultiGCSWriter.java

```diff
@@ -69,23 +70,23 @@ public MultiGCSWriter(Storage storage, String bucket, String baseObjectName,
     this.storage = storage;
     this.bucket = bucket;
     this.baseObjectName = baseObjectName;
-    this.objects = new HashMap<>();
+    this.objects = new LinkedHashMap<>();
```

Comment:
This change is made to make sure that snapshot records landed first in the target. `LinkedHashMap` will provide insert ordering guarantee and since snapshot records come in first, key with the snapshot enabled will be first.

Reply:
if ordering is important it's better to use a List, otherwise it will be easy to break this accidentally in the future.

Reply:
though the key already has snapshot info, it would be more clear to explicitly look for that information to load snapshots first rather than relying on order.

Reply:
Added enum to distinguish between `SNAPSHOT` and `STREAMING` blobs. PTAL.